### PR TITLE
feat(client): files manager interactions

### PR DESCRIPTION
**Pull request by jkergal**
## Ticket

**Title: ETQMembre, je peux intéragir avec des fichiers avec un clic droit**
**Notion Link: https://www.notion.so/jkergal/ETQMembre-je-peux-int-ragir-avec-des-fichiers-avec-un-clic-droit-02beeb7deeaa482ba03d0d3dd2a4a863?pvs=4**

## Description

- creates a context-menu component
- implements file renaming
- implements file deletting
- fixes several state issues
- deletes lagging prettier plugin
- modifies server port to work with OSX that already uses 5000 by default

image
image

## Guidelines

🔗 Link your PRs to their related Notion tickets
✍️ Use the commit convention to your pull request title
📃 If your code affects the build, update `README.md`
🚫 Do not merge a PR until all comments are resolved
🗑 Remove your branch after merging

### CODE REVIEW

File: server/src/models/Project.ts [hunk 1]
```diff
@@ -41,6 +40,16 @@ export class Project extends BaseModel {
...
 
+	@Column()
+	@Field()
+	@IsString()
+	environment: string;
+
+	@Column()
+	@Field()
+	@IsString()
+	main: string;
```
Comment (reynan-dev) [inline server/src/models/Project.ts:51]: main is so generic, we can use **mainFile** or **activeFile**.

Comment (jkergal) [reply server/src/models/Project.ts:51]: yes for mainFile, you right
activeFile is something else btw

**Review by reynan-dev — APPROVED** [review]
Okay, approved. just update `main` to `mainFile`.

File: server/src/models/Project.ts (hunk 1 above)
Comment (jkergal) [reply server/src/models/Project.ts:51]: i will make the modifications in last PR so i won't have to rebase branches

**Pull request merged**